### PR TITLE
feat: forced enable of filter. #2391

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
### Reviewers
**Functional:** 
@seve 

---

## Changes
- Internally forced filter functionality to be enabled. This update is a temporary measure to enable the filter functionality (aiming for an April 27 release) and allows filter flag switches (and this update) to be removed "behind the scenes" (#1718).

## Definition of Done (from ticket)
* Filter functionality is always enabled.
* Filter query string param is ignored.
* Data Portal homepage is updated from "/" to "/datasets".

### CODE REVIEW

File: frontend/src/views/Collections/index.tsx [hunk 1]
```diff
@@ -53,23 +52,10 @@ const COLUMN_ID_RECENCY = "recency";
  */
 const RECENCY = "recency";
 
-/**
- * Gene sets CSV upload functionality, available if gene sets feature flag is enabled
- */
-const AsyncUploadCSV = loadable(
```
Comment (MillenniumFalconMechanic) [inline frontend/src/views/Collections/index.tsx:59]: Removed as per [Slack](https://clevercanary.slack.com/archives/C02CF7MQPGV/p1642655185046100).

File: frontend/next.config.js [hunk 2]
```diff
@@ -64,4 +64,15 @@ module.exports = {
       },
     ];
   },
+
+  // Redirect / to /datasets.
```
Comment (MillenniumFalconMechanic) [inline frontend/next.config.js:68]: Homepage is now `/datasets`.

File: frontend/src/components/Datasets/components/Grid/components/DatasetNameCell/index.tsx [hunk 3]
```diff
@@ -24,7 +24,11 @@ export default function DatasetNameCell({
       {collectionName &&
         (collectionId ? (
           <SubTitle>
-            <LinkCell url={url} value={collectionName} />
+            <LinkCell
+              data-test-id="collection-link"
```
Comment (MillenniumFalconMechanic) [inline frontend/src/components/Datasets/components/Grid/components/DatasetNameCell/index.tsx:28]: Added to support homepage tests.

File: frontend/tests/features/collection/collection.test.ts [hunk 4]
```diff
@@ -91,6 +97,8 @@ describe("Collection", () => {
 async function createCollection(
   collection?: Partial<Collection>
 ): Promise<string> {
+  await goToPage(`${TEST_URL}${ROUTES.MY_COLLECTIONS}`);
```
Comment (MillenniumFalconMechanic) [inline frontend/tests/features/collection/collection.test.ts:100]: Datasets index page does not display "Create Collection" button; must navigate to My Collections.

**Review by seve — APPROVED** [review]
Why force enable filter first and not remove?

File: frontend/src/views/Collections/index.tsx (hunk 1 above)
Comment (seve) [reply frontend/src/views/Collections/index.tsx:61]: broken link for CZI folks

**Comment by MillenniumFalconMechanic** [comment]
> Why force enable filter first and not remove?

Some of the filter on/off switches are fairly intricate and we wanted to remove any time pressure (i.e. the April 27 go-live) from fully removing the switches. @frano-m and I will be focusing on the removal of the switches next week (#1718).

**Pull request merged**